### PR TITLE
Add command-line-api repo.

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

**Comment by crummel** [comment]
This adds a bunch of prebuilts but gets rid of the System.CommandLine ones, which we need to do.  I logged https://github.com/dotnet/source-build/issues/1764 to work through this prebuilts, which I mostly expect to be new ref packages and previously-source-built packages.

File: patches/command-line-api/0001-Exclude-dotnet-suggest-from-source-build.patch [hunk 1]
```diff
@@ -0,0 +1,215 @@
+From eeee907f22d384322975644dae2c1c1e884ea61d Mon Sep 17 00:00:00 2001
+From: Chris Rummel <crummel@microsoft.com>
+Date: Fri, 18 Sep 2020 16:09:49 -0500
+Subject: [PATCH 1/2] Exclude dotnet-suggest from source-build.
```
Comment (dagood) [inline patches/command-line-api/0001-Exclude-dotnet-suggest-from-source-build.patch:4]: Why not build it?

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -37,6 +37,11 @@
       <Uri>https://github.com/dotnet/templating</Uri>
       <Sha>096be35db97669a59b298168b2be30f9b43eab8a</Sha>
     </Dependency>
+    <!-- this should have CPD to Roslyn but Roslyn doesn't have the proper Version.Details.xml entry - XXX issue number here -->
```
Comment (dagood) [inline eng/Version.Details.xml:40]: Fill in?

> XXX issue number here

File: repos/command-line-api.proj [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  <PropertyGroup>
+    <BuildCommandArgs>--restore --build --pack</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) --configuration $(Configuration)</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) --binaryLog</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) -ci</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) $(FlagParameterPrefix)nodereuse $(ArcadeFalseBoolBuildArg)</BuildCommandArgs>
+    <BuildCommand>$(ProjectDirectory)build$(ShellExtension) $(BuildCommandArgs)</BuildCommand>
+
+    <GlobalJsonFile>$(ProjectDirectory)global.json</GlobalJsonFile>
+    <NuGetConfigFile>$(ProjectDirectory)/NuGet.config</NuGetConfigFile>
```
Comment (dagood) [inline repos/command-line-api.proj:14]: Unnecessary `/`?

**Review by dagood — APPROVED** [review]

File: patches/command-line-api/0001-Exclude-dotnet-suggest-from-source-build.patch (hunk 1 above)
Comment (crummel) [reply patches/command-line-api/0001-Exclude-dotnet-suggest-from-source-build.patch:4]: dotnet-suggest isn't used downstream by anything and adds even more prebuilts, so I figured we can skip it for now.

File: eng/Version.Details.xml (hunk 2 above)
Comment (crummel) [reply eng/Version.Details.xml:40]: Logged https://github.com/dotnet/runtime/issues/42637 and updated the comment.

File: repos/command-line-api.proj (hunk 3 above)
Comment (crummel) [reply repos/command-line-api.proj:14]: Fixed.

File: patches/command-line-api/0001-Exclude-dotnet-suggest-from-source-build.patch (hunk 1 above)
Comment (dagood) [reply patches/command-line-api/0001-Exclude-dotnet-suggest-from-source-build.patch:4]: Would be nice to include this in the patch desc (vs. looking at `blame` and navigating to this PR).

**Review by dagood — APPROVED** [review]

**Pull request merged**